v2/ical.NET.UnitTests/DataTypes/RequestStatusTest.cs

```diff
@@ -1,123 +1,127 @@
-﻿using Ical.Net.DataTypes;
+﻿using System.Collections.Generic;
+using Ical.Net.DataTypes;
 using Ical.Net.Interfaces.General;
 using NUnit.Framework;
+using NUnit.Framework.Interfaces;
 
 namespace Ical.Net.UnitTests
 {
-    [TestFixture]
     public class RequestStatusTest
     {
-        [Test]
-        public void InitialObjectState_Constructor_Default()
+        private static IEnumerable<ITestCaseData> ConstructorTestCases() 
         {
-            var requestStatus = new RequestStatus();
+            var emptyRequestStatus = new RequestStatus();
+            yield return new TestCaseData(emptyRequestStatus, null, null, null)
+                .SetName("Default request status constructor has null properties");
 
-            Assert.IsNull(requestStatus.StatusCode);
-            Assert.IsNull(requestStatus.Description);
-            Assert.IsNull(requestStatus.ExtraData);
+            var invalidSerializedValue = new RequestStatus("ABC");
+            yield return new TestCaseData(invalidSerializedValue, null, null, null)
+                .SetName("Constructor with invalid serialised text has null properties");
         }
 
-        [Test]
-        public void InitialObjectState_Constructor_InvalidSerialisedValue() 
+        [Test, TestCaseSource(nameof(ConstructorTestCases))]
+        public void ConstructorTests(RequestStatus incoming, string expectedDescription, string expectedStatusCode, string expectedExtraData) 
         {
-            var requestStatus = new RequestStatus("ABC");
-
-            Assert.IsNull(requestStatus.StatusCode);
-            Assert.IsNull(requestStatus.Description);
-            Assert.IsNull(requestStatus.ExtraData);
+            Assert.AreEqual(incoming.StatusCode, expectedStatusCode);
+            Assert.AreEqual(incoming.Description, expectedDescription);
+            Assert.AreEqual(incoming.ExtraData, expectedExtraData);
         }
 
         [Test, Ignore("Understand what 'valid' values RequestStatusSerializer expects")]
         public void InitialObjectState_Constructor_ValidSerialisedValue() { }
 
-        [Test]
-        public void ToString_ValidInputWithNullStatusCode()
+        private static IEnumerable<ITestCaseData> ToStringTestCases() 
         {
-            var requestStatusA = new RequestStatus();
-            var requestStatusB = new RequestStatus()
+            var emptyRequestStatus = new RequestStatus();
+            yield return new TestCaseData(emptyRequestStatus, ";")
+                .SetName("ToString() for empty request status should return semicolon");
```

Comment:
You are 100% Rian. I wrote the unit tests to validate the current behavior assuming it is the correct one as I am still not familiar enough with the iCalendar RFC specification to know differently. 

By adding the unit tests we _might_ be able to bring into light omissions or errors which can be otherwise hard to spot. 

I will be happy to change the unit tests as you see fit.